src/main/java/in/ashwanthkumar/gocd/slack/jsonapi/Server.java

```diff
@@ -48,17 +48,21 @@ JsonElement getUrl(URL url)
 
         HttpURLConnection request = httpConnectionUtil.getConnection(normalizedUrl);
 
-        // Add in our HTTP authorization credentials if we have them. Favor the API Token
-        // over username/password
+        // Add in our HTTP authorization credentials if we have them.
+        // Favor the API Token over username/password
+        String authHeader = null;
         if (isNotEmpty(mRules.getGoAPIToken())) {
             String bearerToken = "Bearer "
                     + DatatypeConverter.printBase64Binary(mRules.getGoAPIToken().getBytes());
             request.setRequestProperty("Authorization", bearerToken);
         } else if (isNotEmpty(mRules.getGoLogin()) && isNotEmpty(mRules.getGoPassword())) {
             String userpass = mRules.getGoLogin() + ":" + mRules.getGoPassword();
-            String basicAuth = "Basic "
+            authHeader = "Basic "
                     + DatatypeConverter.printBase64Binary(userpass.getBytes());
-            request.setRequestProperty("Authorization", basicAuth);
+        }
+        if (authHeader != null) {
+            request.setRequestProperty("Authorization", authHeader);
```

Comment:
It feels odd to extract only one portion of setting the `Authorization` header out while leaving the other part as is. Can we please revert this line and moving the `Accept` header to above the `if-else` block?

Reply:
You are right, I didn't pushed the right version.